x/derivatives/keeper/margin.go

```diff
@@ -6,6 +6,94 @@ import (
 	"github.com/UnUniFi/chain/x/derivatives/types"
 )
 
+func (k Keeper) AddMargin(ctx sdk.Context, sender sdk.AccAddress, positionId string, amount sdk.Coin) error {
+	position := k.GetPositionWithId(ctx, positionId)
+	if position == nil {
+		return types.ErrPositionDoesNotExist
+	}
+
+	// Send additional margin by sender to the margin manager module account
+	if err := k.SendMarginToMarginManager(ctx, sender, sdk.NewCoins(amount)); err != nil {
```

Comment:
The token transfer should be done latter. In case of an error, the token will be wasted

Reply:
Fixed it.
Please check it out again